bio/salmon/index/wrapper.py

```diff
@@ -6,11 +6,24 @@
 __license__ = "MIT"
 
 from snakemake.shell import shell
+from tempfile import TemporaryDirectory
 
 log = snakemake.log_fmt_shell(stdout=True, stderr=True)
 extra = snakemake.params.get("extra", "")
 
-shell(
-    "salmon index -t {snakemake.input} -i {snakemake.output} "
-    " --threads {snakemake.threads} {extra} {log}"
-)
+decoys = ""
+if "decoys" in snakemake.input.keys():
+    decoys = "--decoys {}".format(snakemake.input["decoys"])
```

Comment:
Maybe a bit cleaner:

```suggestion
decoys = snakemake.input.get("decoys", "")
if "decoys":
    decoys = f"--decoys {decoys}"
```

Reply:
f-strings used here and in salmon-quant wrapper as you suggested, thank you